File: src/org/jgll/grammar/Character.java
package org.jgll.grammar;

import java.util.BitSet;

/**
 * 
 * @author Ali Afroozeh
 *
 */
public class Character extends AbstractTerminal {
	
	private static final long serialVersionUID = 1L;

	private final int c;

	public Character(int c) {
		this.c = c;
	}
	
	public int get() {
		return c;
	}
	
	@Override
	public boolean match(int i) {
		return c == i;
	}
	
	@Override
	public String toString() {
		return getName();
	}

	@Override
	public String getMatchCode() {
		return "I[ci] == " + c;
	}

	@Override
	public int hashCode() {
		return c;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj) {
			return true;
		}
		if (!(obj instanceof Character)) {
			return false;
		}
		Character other = (Character) obj;
		
		return c == other.c;
	}

	@Override
	public String getName() {
		return "[" + (char)c + "]";
	}

	@Override
	public BitSet asBitSet() {
		BitSet set = new BitSet();
		set.set(c);
		return set;
	}

}

File: src-scala/org/jgll/grammar/Character.scala
package org.jgll.grammar

import java.util.BitSet
import org.jgll.grammar.condition.Condition

import collection.mutable._

//remove if not needed
import scala.collection.JavaConversions._

@SerialVersionUID(1L)
class Character(private val c: Int) extends AbstractSymbol with Terminal {

  def get(): Int = c

  override def `match`(i: Int): Boolean = c == i

  override def toString(): String = getName

  override def getMatchCode(): String = "I[ci] == " + c

  override def hashCode(): Int = c

  override def equals(obj: Any): Boolean = {
    if (!(obj.isInstanceOf[Character])) {
      return false
    }
    val other = obj.asInstanceOf[Character]
    c == other.c
  }

  override def getName(): String = "[" + c.toChar + "]"

  override def asBitSet(): BitSet = {
    val set = new BitSet()
    set.set(c)
    set
  }

  override def addConditions(conditions: ListBuffer[Condition]): Terminal = {
    val terminal = new Character(this.c)
    terminal.conditions.++=(this.conditions)
    terminal.conditions.++=(conditions)
    terminal
  }
}
